mesh/mesh.cpp

```
@@ -5250,6 +5257,181 @@ void Mesh::LoadPatchTopo(std::istream &input, Array<int> &edge_to_knot)
+
+         for (j = 0; j < edge1.Size(); j++)
+         {
+            e0 = edges[edge0[j]];
+            e1 = edges[edge1[j]];
+            v0 = edge_to_knot[e0];
+            v1 = edge_to_knot[e1];
+            df = flip*oedge[edge0[j]]*oedge[edge1[j]];
+
+            if ((v0 == notset) && (v1 == notset))
+            {
+               edge_to_knot[e0] = knot;
+               edge_to_knot[e1] = knot;
+               knot++;
+            }
+            else if ((v0 != notset) && (v1 == notset))
+            {
+               edge_to_knot[e1] = (df >= 0 ? -v0-1 : v0);
+            }
+            else if ((v0 == notset) && (v1 != notset))
+            {
+               edge_to_knot[e0] = (df >= 0 ? -v1-1 : v1);
+            }
+         }
+      }
+
+      // Correct assignment, make sure that:
+      // -- corresponding edges within patch point to same knot vector
+      // -- assign the lowest number
+      int corrections;
+
+      for (int pp = 0; pp < 3*GetNE(); pp++)
+      {
+         corrections = 0;
+         for (p = 0; p < GetNE(); p++)
+         {
+            GetElementEdges(p, edges, oedge);
+            for (j = 0; j < edge1.Size(); j++)
+            {
+               e0 = edges[edge0[j]];
+               e1 = edges[edge1[j]];
+               v0 = edge_to_knot[e0];
+               v1 = edge_to_knot[e1];
+               v0 = ( v0 >= 0 ?  v0 : -v0-1);
+               v1 = ( v1 >= 0 ?  v1 : -v1-1);
+               if (v0 != v1)
+               {
+                  corrections++;
+                  if (v0 < v1)
+                  {
+                     edge_to_knot[e1] = (oedge[edge1[j]] >= 0 ? v0 : -v0-1);
+                  }
+                  else if (v1 < v0)
+                  {
+                     edge_to_knot[e0] = (oedge[edge0[j]] >= 0 ? v1 : -v1-1);
+                  }
+               }
+            }
+         }
+         if (corrections == 0) { break; }
+      }
+
+      // Check validity of corrections applied
+      if (corrections > 0 )
```

Should the user be concerned in this case? If the result is not trustworthy, `MFEM_VERIFY` can be used to terminate. In other words, can the code determine a pass/fail result, rather than warning the user?

---

Good suggestion! I have added an mfem_error() as the mesh is problably not fit for use if there are still corrections applied in the last cycle. The error is initiated at the end of the algorithm such that a suggestion for an edge_to_knot mapping is still printed.